src/lib.rs

```diff
@@ -122,14 +122,22 @@ pub enum ArraySize {
     Dynamic,
 }
 
+/// Describes where a struct member is placed.
+#[derive(Clone, Debug, PartialEq)]
+pub enum MemberPlace {
```

Comment:
I am not really sure about this name, would not something like `MemberOrigin` be a little more descriptive? I am just thinking out loud, maybe you had a reason to use the word `Place`